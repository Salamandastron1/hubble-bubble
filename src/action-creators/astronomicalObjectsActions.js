export const starsReceived = astronomicalObjects => ({
  type: 'STARS_RECEIVED',
  astronomicalObjects,
})

export const toggleSelected = id => ({
  type: 'TOGGLE_SELECTED',
  id,
})