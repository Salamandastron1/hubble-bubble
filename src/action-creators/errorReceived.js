export const errorReceived = error => ({
  type: 'ERROR_RECEIVED',
  error
})