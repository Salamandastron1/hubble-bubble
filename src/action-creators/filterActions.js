export const showAll = () => ({
  type: 'SHOW_ALL'
})

export const showGame = () => ({
  type: 'SHOW_GAME',
})

export const showMatches = () => ({
  type: 'SHOW_MATCHES'
})