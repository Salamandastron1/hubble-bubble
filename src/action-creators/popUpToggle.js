export const popUpToggle = () => ({
  type: 'POPUP_TOGGLE',
})