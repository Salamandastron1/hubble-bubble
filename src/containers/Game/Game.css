@import url('https://fonts.googleapis.com/css?family=Roboto');

.game {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.game-form {
  color: white;
  display: grid;
  background: #c17cb3;
  justify-items: center;
  grid-gap: 10px;
  padding: 15px;
  border-radius: 20px;
  font-size: 1.5em;
  font-family: 'Roboto', sans-serif;
}
.bubble-image {
  height: 600px;
  width: 600px;
  border-radius: 50%;
  display: block;
  box-shadow: 0 20px 30px rgba(0, 0, 0, 0.2), inset 0px 10px 30px 5px rgba(255, 255, 255, 1)
}
.submit {
  font-family: 'Titan One', cursive;
  color: white;
  background: #864385;
  border-radius: 30px;
  width: 150px;
  font-size: 1em;
  border: none;
  height: 45px;
}

.submit: hover {
  box-shadow: inset -4px -3px 2px white
}