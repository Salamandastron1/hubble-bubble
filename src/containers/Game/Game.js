import React, { Component } from 'react'
import { connect } from 'react-redux'
import * as starActions from '../../action-creators/astronomicalObjectsActions'
import { makeQuestions } from '../../util/helper'


class Game extends Component {
  constructor() {
    super()
    this.state = {
      selectedAnswer: '',
      answers: [],
      imageSelection: null,
    }
  }

  buttonSelected = (id, randomIndices, subject) => {
    this.setState({
      selectedAnswer: id,
      answers: randomIndices,
      imageSelection: subject,
    })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    this.setState({
      selectedAnswer: '',
      answers: []
    })
  }

  render () {
    const { chooseAnswer, astronomicalObjects, isLoading } = this.props;
    const { selectedAnswer, answers, imageSelection } = this.state;
    let randomIndices;
    let subject;
    if(answers.length > 1) {
      randomIndices = answers
      subject = imageSelection
    } else {
      randomIndices = makeQuestions(astronomicalObjects)
      subject = Math.floor(Math.random() * Math.floor(randomIndices.length - 1))
    }
    const randomAnswers = randomIndices.map(number => {
      return (            
        <div key={astronomicalObjects[number].id}>
            <input 
              type='radio' 
              name='name' 
              value={astronomicalObjects[number].name} 
              checked={astronomicalObjects[number].id === selectedAnswer}
              onChange={() => this.buttonSelected(astronomicalObjects[number].id, randomIndices, subject)}/>
            <label htmlFor={astronomicalObjects[number].name}>{astronomicalObjects[number].name}</label>
        </div>
      )
    })
    if(isLoading) {
      return (
        <div className='spinner'>
          <div className='star1'></div>
          <div className='star2'></div>
        </div>
      )
    } else {
      return (
        <section>
          <img height='400' width='400'src={astronomicalObjects[randomIndices[subject]].image_files} />
            <form
              onSubmit={this.handleSubmit}>
            <fieldset>
              <legend>Which name matches the picture?</legend>
                {randomAnswers}
            </fieldset>
            <input 
              type='submit' 
              value='Submit'/>
          </form>
        </section>
      )
    }
  }
}

export const mapStateToProps = state => ({
  astronomicalObjects: state.astronomicalObjects,
  isLoading: state.isLoading
})

export const mapDispatchToProps = dispatch => ({
  chooseAnswer: (id) => dispatch(starActions.toggleSelected(id))
})
export default connect(mapStateToProps, mapDispatchToProps)(Game)
