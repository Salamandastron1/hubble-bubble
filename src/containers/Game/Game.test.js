import { Game } from './Game'
import { shallow } from 'enzyme'
import { mapStateToProps, mapDispatchToProps} from './Game'
import React from 'react'
import { popUpToggle } from '../../action-creators/popUpToggle'
import configureMockStore from 'redux-mock-store'


describe('Game', () => {
  let mockStore;
  let initialState;
  let store;
  let wrapper;
  beforeEach(() => {
    mockStore = configureMockStore()
    initialState = {movies: []}
    store = mockStore(initialState);
    wrapper = shallow(<Game astronomicalObjects={
      [{id: 1, image_files:'meow', name: 'class'},{id: 2, image_files:'meow', name: 'name'},{id: 3, image_files:'meow', name: 'name'},{id: 4, image_files:'meow', name: 'name'},{id: 5, image_files:'meow', name: 'name'}
      ]}
      togglePopUp={() => jest.fn()} 
      isLoading={false}/>)
  })
  describe('mapStateToProps', () => {
    it('should return a an object with state inside of it', () => {
      const desired = {
        astronomicalObjects: [{id: 4, name: 'name',},{id: 5, name: 'meow'}],
        isLoading: true,
      }
      const mockState = {
        error: 'haserrored',
        astronomicalObjects: [{id: 4, name: 'name',},{id: 5, name: 'meow'}],
        filter: 'SHOW_ALL',
        isLoading: true,
      }
      const result = mapStateToProps(mockState)

      expect(result).toEqual(desired)
    })
  })
  it('should call handleSubmit on submision', () => {
    const mockEvent = {
      preventDefault: jest.fn()
    }
    const mockState = {
      "answers": [1, 3, 2, 0], 
      "correct": false, 
      "imageSelection": 3, 
      "selectedAnswer": 2
    }

    wrapper.setState({
      selectedAnswer: 2,
      answers: [1, 3 , 2, 0],
      imageSelection: 3,
    })
    wrapper.find('form').simulate('submit', mockEvent)

    expect(wrapper.state()).toEqual(mockState)
  })
  it('should matcht the state when a radio button is pressed', () => {
    const mockEvent = {
      preventDefault: jest.fn()
    }
    const mockState = {"answers": [2, 1, 0, 3], "correct": false, "imageSelection": 2, "selectedAnswer": 1}

    wrapper.find('.class').simulate('change', mockEvent)
    expect(wrapper.state().selectedAnswer).toEqual(1)
  })
  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params', () => {
      const mockDispatch = jest.fn()
      const mappedDispatch = mapDispatchToProps(mockDispatch)

      mappedDispatch.togglePopUp()

      expect(mockDispatch).toHaveBeenCalledWith(popUpToggle())
    })
  })
})