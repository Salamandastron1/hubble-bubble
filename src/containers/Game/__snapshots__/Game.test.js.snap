// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should matcht the snapshot 1`] = `
<section
  className="game"
>
  <div
    className="bubble-image"
    style={
      Object {
        "backgroundImage": "url(meow)",
      }
    }
  />
  <form
    className="game-form"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Which name matches the picture?
      </legend>
      <div
        key="4"
      >
        <input
          checked={false}
          className="name"
          name="name"
          onChange={[Function]}
          type="radio"
          value="name"
        />
        <label
          htmlFor="name"
        >
          name
        </label>
      </div>
      <div
        key="1"
      >
        <input
          checked={false}
          className="class"
          name="name"
          onChange={[Function]}
          type="radio"
          value="class"
        />
        <label
          htmlFor="class"
        >
          class
        </label>
      </div>
      <div
        key="2"
      >
        <input
          checked={false}
          className="name"
          name="name"
          onChange={[Function]}
          type="radio"
          value="name"
        />
        <label
          htmlFor="name"
        >
          name
        </label>
      </div>
      <div
        key="3"
      >
        <input
          checked={false}
          className="name"
          name="name"
          onChange={[Function]}
          type="radio"
          value="name"
        />
        <label
          htmlFor="name"
        >
          name
        </label>
      </div>
    </fieldset>
    <input
      className="submit"
      type="submit"
      value="Submit"
    />
  </form>
</section>
`;

exports[`Game should render the state if answers has a length greater than 1 1`] = `
<section
  className="game"
>
  <div
    className="bubble-image"
    style={
      Object {
        "backgroundImage": "url(meow)",
      }
    }
  />
  <form
    className="game-form"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Which name matches the picture?
      </legend>
      <div
        key="2"
      >
        <input
          checked={false}
          className="name"
          name="name"
          onChange={[Function]}
          type="radio"
          value="name"
        />
        <label
          htmlFor="name"
        >
          name
        </label>
      </div>
      <div
        key="4"
      >
        <input
          checked={false}
          className="name"
          name="name"
          onChange={[Function]}
          type="radio"
          value="name"
        />
        <label
          htmlFor="name"
        >
          name
        </label>
      </div>
      <div
        key="3"
      >
        <input
          checked={false}
          className="name"
          name="name"
          onChange={[Function]}
          type="radio"
          value="name"
        />
        <label
          htmlFor="name"
        >
          name
        </label>
      </div>
      <div
        key="1"
      >
        <input
          checked={false}
          className="class"
          name="name"
          onChange={[Function]}
          type="radio"
          value="class"
        />
        <label
          htmlFor="class"
        >
          class
        </label>
      </div>
    </fieldset>
    <input
      className="submit"
      type="submit"
      value="Submit"
    />
  </form>
</section>
`;

exports[`Game should return a different render if isLoading is True 1`] = `
<div
  className="spinner"
>
  <div
    className="star1"
  />
  <div
    className="star2"
  />
</div>
`;
